File: SEP490_Ezstay/RentalPostsAPI/DTO/Response/ApiResponse.cs
namespace RentalPostsAPI.DTO.Response;

public class ApiResponse<T>
{
    public bool IsSuccess { get; set; }
    public string? Message { get; set; }
    public T? Data { get; set; }

    public static ApiResponse<T> Success(T data, string? message = null) => new()
    {
        IsSuccess = true,
        Data = data,
        Message = message
    };

    public static ApiResponse<T> Fail(string message) => new()
    {
        IsSuccess = false,
        Message = message
    };
}

File: SEP490_Ezstay/AccountAPI/AccountAPI/DTO/Response/ApiResponse.java
package AccountAPI.DTO.Response;

public class ApiResponse<T> {
	public final boolean _success;
	public final String _message;
	public final T _data;

	public ApiResponse<T> Ok(T aData, String aMessage) {
		throw new UnsupportedOperationException();
	}

	public ApiResponse<T> Fail(String aMessage) {
		throw new UnsupportedOperationException();
	}

	public boolean isSuccess() {
		return this._success;
	}

	public String getMessage() {
		return this._message;
	}

	public T getData() {
		return this._data;
	}
}